src/Bridge/Doctrine/Orm/Extension/FilterExtension.php

```diff
@@ -60,12 +61,25 @@ public function applyToCollection(QueryBuilder $queryBuilder, QueryNameGenerator
             return;
         }
 
+        $orderFilter = null;
+
         foreach ($resourceFilters as $filterId) {
             $filter = $this->getFilter($filterId);
             if ($filter instanceof FilterInterface) {
+                // Apply the OrderFilter after every other filter to avoid an edge case where OrderFilter would do a LEFT join instead of an INNER join
+                if ($filter instanceof OrderFilter) {
+                    $orderFilter = $filter;
+                    continue;
+                }
+
                 $context['filters'] = $context['filters'] ?? [];
                 $filter->apply($queryBuilder, $queryNameGenerator, $resourceClass, $operationName, $context);
             }
         }
+
+        if (null !== $orderFilter) {
+            $context['filters'] = $context['filters'] ?? [];
+            $orderFilter->apply($queryBuilder, $queryNameGenerator, $resourceClass, $operationName, $context);
+        }
     }
```

Comment:
This is the easiest fix I've found, let me know how you feel about it :) (@meyerbaptiste I think it's just fine like this, no need to get more complicated)